virttest/libvirt_xml/devices/hostdev.py

```diff
@@ -41,15 +49,34 @@ def new_source(self, **dargs):
             new_address = new_one.new_untyped_address(**dargs)
             new_one.untyped_address = new_address
         if self.hostdev_type == 'scsi':
-            new_one.adapter_name = dargs.pop("adapter_name", None)
-            new_address = new_one.new_untyped_address(**dargs)
-            new_one.untyped_address = new_address
+            if dargs.get("adapter_name"):
```

Comment:
Here you have more strict condition for assigning value to  new_one.adapter_name  compared with old one. 
After your change, it need at least the key 'adapter_name' exist in dict, otherwise the assignment will not be executed.
Are you sure that you really need this change?

Reply:
yes, that's intentional, the original assigment always add a adapter name to scsi hostdev xml, as follow:
`new_one.adapter_name = dargs.pop("adapter_name", None)`

But actually the adapter name is not necessary for scsi hostdev, such as follow:
`  <hostdev mode='subsystem' type='scsi'>
    <source protocol='iscsi' name='iqn.2014-08.com.example:iscsi-nopool/1'>
      <host name='example.com' port='3260'/>
      <auth username='myuser'>
        <secret type='iscsi' usage='libvirtiscsi'/>
      </auth>
    </source>
    <address type='drive' controller='0' bus='0' target='0' unit='0'/>
  </hostdev>`

And IIRC, the npiv devel also add code to do something as follow, which does't need the adapter name either.
`
<hostdev mode='subsystem' type='scsi'>
    <source wwnn=xxx wwpn=xxx>
  </hostdev>
`
And in above two situations, the original code will always generate a incorrect hostdev xml with unnecessary adpater name attribute.